cmake/jsoncpp.cmake

```diff
@@ -35,6 +35,7 @@ ExternalProject_Add(jsoncpp-project
                -DJSONCPP_WITH_TESTS=OFF
                -DJSONCPP_WITH_PKGCONFIG_SUPPORT=OFF
                -DCMAKE_CXX_FLAGS=${JSONCPP_EXTRA_FLAGS}
+               -DCMAKE_BUILD_TYPE=Release
     # Overwrite build and install commands to force Release build on MSVC.
     BUILD_COMMAND cmake --build <BINARY_DIR> --config Release
     INSTALL_COMMAND cmake --build <BINARY_DIR> --config Release --target install
```

Comment:
Isn't this `--config Release` override the args above?

Reply:
@axic, nope, the --config part is for stage 2 (build) and stage 3 (install), but not for stage 1 (bootstrap). That is *exactly* why it took me so long to catch. :-)

Reply:
So CMake ignores `CMAKE_BUILD_TYPE` variable when generating for Visual Studio because it's multi-configuration generator and the config is requested with `--config Release` (or via dropdown menu in VS).

I expect the jsoncpp's CMake script incorrectly checks `CMAKE_BUILD_TYPE` on Windows and that's why this fix works.

Reply:
Do we want to leave a comment in there explaining this? Also anybody feels like checking jsoncpp's cmake file and fixing it? :)

Reply:
@chfast so I checked the build files of jsoncpp, and it's not reading the ``CMAKE_BUILD_TYPE`` var when compiling on Windows platform. So I am inclined to believe that VS2017 (at least) is interpreting this value when using the native builtin cmake support (FYI: I do not generate .sln files with cmake but open the `CMakeLists.txt` as solution file instead.)

I'll close this ticket however, because the other PR makes this PR redundant. Thanks anyways. :-)